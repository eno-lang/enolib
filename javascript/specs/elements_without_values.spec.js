import { parse } from '..';
import { unpack } from './unpack.js';

const sample = `
empty:
leer:

-

-

-

nichts:
nada:

# nothing
## none
void:
emptyness:
-- leere
-- leere

# ningun
absence:
-

-
-


non:
end:
`;

describe('Elements without values', () => {
    it('correctly parses', () => {
        const document = parse(sample);
        expect(unpack(document)).toMatchSnapshot();
    });
});
