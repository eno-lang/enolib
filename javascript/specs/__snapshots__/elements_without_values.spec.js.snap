// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Elements without values correctly parses 1`] = `
Object {
  "elements": Array [
    Object {
      "key": "empty",
      "type": "field",
    },
    Object {
      "items": Array [
        Object {},
        Object {},
        Object {},
      ],
      "key": "leer",
      "type": "field",
    },
    Object {
      "key": "nichts",
      "type": "field",
    },
    Object {
      "key": "nada",
      "type": "field",
    },
    Object {
      "elements": Array [
        Object {
          "elements": Array [
            Object {
              "key": "void",
              "type": "field",
            },
            Object {
              "key": "emptyness",
              "type": "field",
            },
            Object {
              "key": "leere",
              "type": "embed",
            },
          ],
          "key": "none",
          "type": "section",
        },
      ],
      "key": "nothing",
      "type": "section",
    },
    Object {
      "elements": Array [
        Object {
          "items": Array [
            Object {},
            Object {},
            Object {},
          ],
          "key": "absence",
          "type": "field",
        },
        Object {
          "key": "non",
          "type": "field",
        },
        Object {
          "key": "end",
          "type": "field",
        },
      ],
      "key": "ningun",
      "type": "section",
    },
  ],
  "key": null,
  "type": "section",
}
`;
